cmd/kubeless/function/function.go

```diff
@@ -157,47 +157,49 @@ func getFunctionDescription(cli kubernetes.Interface, funcName, ns, handler, fro
 		function.Spec.Handler = handler
 	}
 
-	if fromURL != "" && file != "" {
-		err := errors.New("either --from-url or --from-file should be provided, not both")
-		return nil, err
-	}
-
-	if fromURL != "" {
-		resp, err := http.Get(fromURL)
-		if err != nil {
-			return nil, err
-		}
-		defer resp.Body.Close()
-
-		functionBytes, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return nil, err
-		}
-
-		checksum, err := getSha256(functionBytes)
-		if err != nil {
-			return nil, err
-		}
-		function.Spec.Checksum = checksum
+	if file != "" {
 
-		function.Spec.FunctionContentType = "url"
-		function.Spec.Function = fromURL
-	}
+		if strings.Index(file, "http://") == 0 || strings.Index(file, "https://") == 0 {
+			functionURL, err := url.Parse(file)
+			if err != nil {
+				return nil, err
+			}
+			resp, err := http.Get(functionURL.String())
+			if err != nil {
+				return nil, err
+			}
+			defer resp.Body.Close()
+
+			functionBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			function.Spec.FunctionContentType = "url"
+			// strip any query params to determine if the file is a zip
+			if path.Ext(strings.Split(functionURL.String(), "?")[0]) == ".zip" {
+				function.Spec.FunctionContentType += "+zip"
+			}
+			function.Spec.Function = functionURL.String()
+			function.Spec.Checksum, err = getSha256(functionBytes)
+			if err != nil {
+				return nil, err
+			}
 
-	if file != "" {
-		functionBytes, err := ioutil.ReadFile(file)
-		if err != nil {
-			return nil, err
-		}
-		function.Spec.FunctionContentType = getContentType(file, functionBytes)
-		if function.Spec.FunctionContentType == "text" {
-			function.Spec.Function = string(functionBytes)
 		} else {
-			function.Spec.Function = base64.StdEncoding.EncodeToString(functionBytes)
-		}
-		function.Spec.Checksum, err = getFileSha256(file)
-		if err != nil {
-			return nil, err
+			functionBytes, err := ioutil.ReadFile(file)
+			if err != nil {
+				return nil, err
+			}
+			function.Spec.FunctionContentType = getContentType(file, functionBytes)
+			if function.Spec.FunctionContentType == "text" {
+				function.Spec.Function = string(functionBytes)
+			} else {
+				function.Spec.Function = base64.StdEncoding.EncodeToString(functionBytes)
+			}
+			function.Spec.Checksum, err = getFileSha256(file)
```

Comment:
I would create another function here that given a content type it returns the content and its checksum:

```
func parseContent(content, type) (string, string, error)
```

that function would read a file in case the type is `text` do the `base64` in case it is a zip or download the file in case it is an URL. That way you can reuse it for the dependencies file.